core/core-configuration-layer.el

```diff
@@ -475,20 +475,28 @@ Properties that can be copied are `:location', `:step' and `:excluded'."
 (defun configuration-layer//copy-template (name template &optional layer-dir)
   "Copy and replace special values of TEMPLATE to layer string NAME.
 If LAYER_DIR is nil, the private directory is used."
-  (let ((src (concat configuration-layer-template-directory
-                     (format "%s.template" template)))
-        (dest (if layer-dir
-                  (concat layer-dir "/" (format "%s" template))
-                (concat (configuration-layer//get-private-layer-dir name)
-                        (format "%s" template)))))
-    (copy-file src dest)
-    (find-file dest)
-    (save-excursion
-      (goto-char (point-min))
-      (let ((case-fold-search nil))
-        (while (re-search-forward "%LAYERNAME%" nil t)
-          (replace-match name t))))
-    (save-buffer)))
+  (cl-flet ((substitute
+             (old new)
+             (let ((case-fold-search nil))
+               (save-excursion
+                 (goto-char (point-min))
+                 (while (search-forward old nil t)
+                   (replace-match new t))))))
+
+    (let ((src (concat configuration-layer-template-directory
+                       (format "%s.template" template)))
+          (dest (if layer-dir
+                    (concat layer-dir "/" (format "%s" template))
+                  (concat (configuration-layer//get-private-layer-dir name)
+                          (format "%s" template)))))
+      (copy-file src dest)
+      (find-file dest)
+
+      (substitute "%LAYER_NAME%" name)
+      (substitute "%USER_FULL_NAME%" (user-full-name))
```

Comment:
The value of `user-full-name` is empty on my computer, it may be nice to check this and use a default value instead (Sylvain Benner & Contributors, Unknown, `user-login-name`, … ?) to avoid it being missed.